FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -774,6 +774,14 @@ namespace FEXCore::Context {
   }
 
   uintptr_t ContextImpl::CompileBlock(FEXCore::Core::CpuStateFrame *Frame, uint64_t GuestRIP, uint64_t MaxInst) {
+#ifdef _M_ARM_64EC
```

Comment:
This block of code should probably be move at least after `FEXCORE_PROFILE_SCOPED` if not after pulling the thread object or after the CodeInvalidationMutex gets locked.